### DISCUSSION TITLE
Pango-1.0-0.dll missing attributes

### DISCUSSION THREAD

**Discussion by Raizencorp** [pyinstaller/pyinstaller#7728] [Help] [answered]
Hi,
I'm having an issue with the pango file because it seems he is not up to date. I'm using Weasyprint to convert some html to pdf and it need at least the version 1.44 to work.
Here is my ticket on their side: https://github.com/Kozea/WeasyPrint/issues/1898

How can i fix this myself ? I cannot find a dll file of pango v1.44 on the net. And i never compiled a dll file before. (I'm saying this because i found the source code but no dll)

How does Pyinstaller provide this file anyway ?

Thanks for your time/clarification :)

**Comment by bwoodsend** [comment, marked as answer]
You're not giving us much information to work with here so that leaves you mostly on your own. What I can tell you is compiling any DLL from source is not the answer. If what's said [here](https://github.com/Kozea/WeasyPrint/issues/1898#issuecomment-1598551173) is the case and you've landed with an old version of pango then that means you had two pango libraries and PyInstaller found the wrong one. If you look it at `build/yourapplicationname/Analysis-00.toc`, there should be an entry with pango in it telling you where it found the old one. I'm guessing that the new pango in not findable in `PATH` so the fix would most likely be to find the new one, prepend it to `PATH` then rerun PyInstaller. I'm not feeling very confident in that claim though.

**Reply by Raizencorp** [reply]
To be honest i don't know what kind of other information i could give. The ticket on weasyprint have all the message error. Maybe if you want some specific.
But to be honest this file is like a gold mine ! And he is using the dll from Tesseract (allow to read text from picture)

**Reply by rokm** [reply]
PyInstaller collects DLLs from your build system. So if the DLL was found, and is of incorrect version, then like Brénainn said, you likely have multiple copies of that DLL on your system, and wrong (old) copy was collected.

So to start figuring what went wrong, open `build/<name>/Analysis-00.toc` file with text editor, and find entry for `pango-1.0-0.dll` to see where it was collected from.

**Reply by Raizencorp** [reply]
Thanks guys it really help a lot. Now i just need to give the Fontconfig file to pyintsller to make everything work
